src/MutexInterface.php

```diff
@@ -33,11 +33,20 @@ interface MutexInterface
      *
      * @param int $timeout Time (in seconds) to wait for lock to be released. Defaults to zero meaning that method
      * will return false immediately in case lock was already acquired.
+     *
+     * @return bool Whether a lock is acquired.
      */
     public function acquire(int $timeout = 0): bool;
 
     /**
      * Releases a lock.
      */
     public function release(): void;
+
+    /**
+     * Checks whether a lock is released.
+     *
+     * @return bool Whether a lock is released.
+     */
+    public function isReleased(): bool;
```

Comment:
Tell me please a case when we need to have it?

Reply:
When you have many mutexes at the same time (event loop) or you pass the mutex further, where code does not know the mutex was released or not yet. Also this method uses by `__destruct` method to autorelease mutex on shutdown.

Reply:
If we make release() idempotent you won't need to have this method.

Reply:
You mean always try release on each `release()` call and if it fails, be silent?

Reply:
> If we make release() idempotent you won't need to have this method.

I agree. I redid it.